adivinanzas.py

```diff
@@ -7,8 +7,7 @@ def leer(nombre_archivo):
     archivo = open(nombre_archivo, "r")
     
     # Ahora tenemos todo el contenido del archivo en la variable "cuerpo".
-    cuerpo = unicode(archivo.read())
-    cuerpo.decode("utf-8")
+    cuerpo = str(archivo.read())
```

Comment:
Why is `str` necessary?  Doesn't open with mode `r` guarantee str?

Reply:
Didn't see the mode `r`, sorry.

You are absolutely right, str is not at all necessary here.

In Python 3 when we use in text mode `r`, Python will parse the file according to the text encoding you give it (or if you don't give one, a platform-dependent default), and `read()` will give you an `str`.

Since python3, The default encoding for Python source code is UTF-8 (also mentioned in the `python 3` documentation)